### PR TITLE
Add Lisk Mainnet to the Superchain Registry

### PR DESCRIPTION
**Pull request by matjazv**
# Adding a New Chain
This PR adds Lisk Mainnet to the registry.

## Checklist
- [x] I have declared the chain at the appropriate [Superchain Level](../docs/glossary.md#superchain-level-and-rollup-stage).
- [x] I have run `just validate 1135` locally to ensure all local validation checks pass. 
- [x] I have run `just codegen` to ensure that the chainlist and other generated files are up-to-date and include my chain.

## Additional Information
To enable the `just add-chain` and `just codegen` commands to function correctly, I had to make a few changes:

1. [op-geth (config.go)](https://github.com/ethereum-optimism/op-geth/blob/optimism/params/config.go#L35)
- Adding Lisk Mainnet chain ID into the constants:
`LiskMainnetChainID      = 1135`
- The same will eventually apply to Lisk Sepolia.

2. [op-geth (superchain.go)](https://github.com/ethereum-optimism/op-geth/blob/optimism/params/superchain.go#L86):
- The Lisk Mainnet network has custom values for the following variables, necessitating a new `case` statement:
```
case LiskMainnetChainID:
    out.Optimism.EIP1559Elasticity = 20
    out.Optimism.EIP1559Denominator = 1000
    out.Optimism.EIP1559DenominatorCanyon = newUint64(1000)
```
- The same will eventually apply to Lisk Sepolia.

3. [check-genesis.go](https://github.com/ethereum-optimism/superchain-registry/blob/main/add-chain/cmd/check-genesis.go#L44-%23L45):
- Currently, `EIP1559DenominatorCanyon` is hardcoded to `250`, which does not align with Lisk Mainnet (and Sepolia).
- For now, I have commented out these two lines.

Additionally, the `just validate` command also fails for some of the unit tests.

### CODE REVIEW

**Comment by matjazv** [comment]
With all recent updates, changes inside Additional Information are not needed anymore to successfully execute `just add-chain` command.

**Comment by geoknee** [comment]
> With all recent updates, changes inside Additional Information are not needed anymore to successfully execute `just add-chain` command.

Excellent!

File: validation/genesis/validation-inputs/1135/deploy-config.json [hunk 1]
```diff
@@ -16,16 +16,16 @@
...
-  "baseFeeVaultRecipient": "0xdA6e5640aFB2ED212Ba3a6fd83076e2ad3daD185",
-  "l1FeeVaultRecipient": "0xdA6e5640aFB2ED212Ba3a6fd83076e2ad3daD185",
-  "sequencerFeeVaultRecipient": "0xdA6e5640aFB2ED212Ba3a6fd83076e2ad3daD185",
-  "baseFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
-  "l1FeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
-  "sequencerFeeVaultMinimumWithdrawalAmount": "0x8ac7230489e80000",
-  "baseFeeVaultWithdrawalNetwork": 0,
-  "l1FeeVaultWithdrawalNetwork": 0,
-  "sequencerFeeVaultWithdrawalNetwork": 0,
-  "proxyAdminOwner": "0xdA6e5640aFB2ED212Ba3a6fd83076e2ad3daD185",
```
Comment (geoknee) [inline validation/genesis/validation-inputs/1135/deploy-config.json:28]: Can you please double check these changes? The automatic genesis validation check `validate-genesis-allocs` uses this data, and the existing values were enough to make it pass, the new values cause it to fail.

Comment (matjazv) [reply validation/genesis/validation-inputs/1135/deploy-config.json:28]: I have confirmed the changes with our RaaS provider, which were made when ownership of the chain was transferred from the deployer EOA to a multisig. The values were also adjusted (decreased) for fee withdrawals. As a result, the new values are correct.

This was not the case for Lisk Sepolia so genesis validation check passes inside its PR.

Comment (matjazv) [reply validation/genesis/validation-inputs/1135/deploy-config.json:28]: All those changes were made around 22nd July 2024.

Comment (matjazv) [reply validation/genesis/validation-inputs/1135/deploy-config.json]: I've reverted the changes inside `deploy-config.json` file.

**Comment by sbvegan** [comment]
Hey @matjazv, can you please rebase this branch and rerun `just codegen` ?

**Comment by matjazv** [comment]
> Hey @matjazv, can you please rebase this branch and rerun `just codegen` ?

I've rebased the branch and reran `just codegen` and now everything seems to be fine.

**Review by sbvegan — APPROVED** [review]

**Review by bitwiseguy — APPROVED** [review]

**Pull request merged**